hack/v1migration/phase2.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/sh
+
+set -e
+
+if [ $# != 1 ]; then
+	echo "USAGE: $0 [WORKDIR]"
+	exit 1
+fi
+
+WORKDIR=$1
+echo "Using workdir: $WORKDIR"
+
+oc project hive
+
+oc scale -n hive deployment.v1.apps/hive-operator --replicas=0
+oc scale -n hive deployment.v1.apps/hive-controllers --replicas=0
+oc scale -n hive deployment.v1.apps/hiveadmission --replicas=0
+
+if [[ ! -d "$WORKDIR" ]]
+then
+    echo "Directory $WORKDIR does not exist."
+    exit 1
+fi
+
+
+HIVE_TYPES=( checkpoints clusterdeployments clusterdeprovisionrequests clusterimagesets clusterprovisions clusterstates dnsendpoints dnszones hiveconfig selectorsyncidentityprovider selectorsyncset syncidentityprovider syncsetinstance syncset )
+for i in "${HIVE_TYPES[@]}"
+do
+	:
+	echo "Deleting all ${i}"
+	oc delete ${i}.hive.openshift.io -A --all
```

Comment:
Before deleting, I need two checks.
(1) The user must acknowledge that they meant to run the delete.
(2) We must check that the resource stored in yaml and matches the yaml.

Reply:
(1) makes sense, what do you mean by (2) and why?

Reply:
I think I figured it out, probably to see if anything restored finalizers on the objects, in which case we're in trouble.

Agreed, but we should probably skip this in phase 1.

Reply:
A safety check to make sure that we have actually stored the resource to file before we delete it. In the extreme, I am thinking of a case where phase 2 is accidentally executed before phase 1.